server/chat/backend/agent/tools/alert_payload_tool.py

```
@@ -0,0 +1,155 @@
+                )
+                payload_row = cursor.fetchone()
+
+                # Fallback: query by user + most recent (handles providers where
+                # source_alert_id is not the event table row ID, e.g. Grafana CRC)
+                if not payload_row or not payload_row[0]:
+                    cursor.execute("SELECT org_id FROM incidents WHERE id = %s", (incident_id,))
+                    org_row = cursor.fetchone()
+                    if org_row:
+                        cursor.execute(
+                            f"SELECT payload FROM {table} WHERE user_id = %s AND org_id = %s "
+                            f"ORDER BY received_at DESC LIMIT 1",
```

Ruff (0.15.14)

[error] Possible SQL injection vector through string-based query construction

(S608)